src/Form/Form.js

```diff
@@ -734,18 +738,31 @@ class Form {
         return (!this.touched.has(input) && !input.classList.contains('ddg-autofilled'))
     }
 
+    /**
+     * Skip overridding values that the user provided if:
+     * - we're autofilling non credit card type and,
+     * - it's a previously filled input or,
+     * - it's a select input that was already "touched" by the user.
+     *  */
+    shouldSkipInput (input, dataType) {
+        // input is of select type and has been touched by the user
+        const isTouchedSelect = input.nodeName === 'SELECT' && this.touched.has(input)
+        // input is non-empty or non-active
+        const isPreviouslyFilledInput = input.value !== '' && this.activeInput !== input
+        // creditCards always override
+        const isNotCreditCardType = dataType !== 'creditCards'
+
+        return isNotCreditCardType && (isTouchedSelect || isPreviouslyFilledInput)
+    }
+
     autofillInput (input, string, dataType) {
         // Do not autofill if it's invisible (select elements can be hidden because of custom implementations)
         if (input instanceof HTMLInputElement && !isPotentiallyViewable(input)) return
         // Do not autofill if it's disabled or readonly to avoid potential breakage
         if (!canBeInteractedWith(input)) return
 
-        // Don't override values the user provided, unless it's the focused input or we're autofilling creditCards
-        if (
-            dataType !== 'creditCards' && // creditCards always override, the others only when we're focusing the input
-            input.nodeName !== 'SELECT' && input.value !== '' && // if the input is not empty
-            this.activeInput !== input // and this is not the active input
-        ) return // do not overwrite the value
+        if (this.shouldSkipInput(input, dataType)
+        ) return
```

Comment:
might as well drop this onto a single line now, no need to preserve the formatting from before :)